libs/langchain-community/src/vectorstores/cassandra.ts

```diff
@@ -158,20 +181,22 @@ export class CassandraStore extends VectorStore {
    * Method to search for vectors that are similar to a given query vector.
    * @param query The query vector.
    * @param k The number of similar vectors to return.
-   * @param filter
+   * @param filter Optional filter to be applied as a WHERE clause.
+   * @param includeEmbedding Whether to include the embedding vectors in the results.
    * @returns Promise that resolves with an array of tuples, each containing a Document and a score.
    */
   async similaritySearchVectorWithScore(
     query: number[],
     k: number,
-    filter?: WhereClause
+    filter?: WhereClause,
+    includeEmbedding?: boolean
```

Comment:
Instead of adding another parameter, factor out query logic into a helper function like this:

https://github.com/langchain-ai/langchainjs/blob/main/libs/langchain-community/src/vectorstores/supabase.ts#L151

Then we can call that from both this method and the MMR method below

Reply:
Hmm. I see what you're saying here, but consider:

1. The query logic itself is in one place; 
2. The MMR function needs to do a similaritySearchVectorWithScore - it just needs the vectors to do its thing; 
3. A caller to similaritySearchVectorWithScore probably does not want the vectors back (why spend time/bandwidth processing things you don't need), but there may be a reason they do, so this gives the caller an option. 

As a new parameter it is optional (for backwards compatibility) and defaults to the prior behaviour... 

That said I can refactor to a separate helper function, but I'd like to keep the new parameter too :)

Reply:
Thanks - messing with the base class signature (even adding more params) has burned us in the past.